.github/workflows/auto-merge.yaml

```diff
@@ -0,0 +1,42 @@
+name: auto-merge
+
+on:
+  workflow_call:
+    secrets:
+      AUTOMATION_TOKEN:
+        required: true
+
+  pull_request_review:
+    types:
+      - submitted
+  check_suite:
+    types:
+      - completed
+  status: {}
+
+jobs:
+  auto-approve:
+    name: auto-approve prs by boldlinksig
+    runs-on: ubuntu-latest
+    steps:
+      - name: auto-approve
+        uses: hmarr/auto-approve-action@v3
+        if: github.actor == 'ndegwajohn'
+        with:
+          github-token: "${{ secrets.GITHUB_TOKEN }}" 
+  auto-merge:
+    runs-on: ubuntu-latest
+    needs: auto-approve
+    steps:
+      - id: auto-merge
+        name: auto-merge
+        uses: "pascalgn/automerge-action@v0.15.5"
+        if: github.actor == 'ndegwajohn'
+        env:
+          GITHUB_TOKEN: "${{ secrets.AUTOMATION_TOKEN }}"  
+          MERGE_LABELS: "automerge"
+          MERGE_METHOD: "squash"
+          MERGE_RETRIES: "6"
+          MERGE_RETRY_SLEEP: "10000"
```

Comment:
Are these seconds? That would be 16 hours, which means that this would be retried for 4 days at 16 hours intervals before failing; I'm really struggling to understand the logic and purpose... please tell

Reply:
The time is in milliseconds, so if other checks are still ongoing it will try again after 10 seconds. It will do 6 tries before it exits with a fail or passes successfully. Max total time it may take is 1 minute.

Reply:
Looks to me this is to be run at the end of the other workflows, you must set priority otherwise you could auto-merge broken code and we don't want that... Think of the order and dependencies your workflows have, you could have one single workflow on the executing repo no matter the order here.

This retry is bad logic.